crux/src/Crux/Goal.hs

```diff
@@ -328,24 +322,22 @@ dispatchSolversOnGoalAsync mtimeoutSeconds adapters withAdapter = do
 -- satisfiability checking and goal discharge.
 proveGoalsOnline ::
   ( sym ~ OnlineBackend s solver fs
-  , ast ~ SimError
   , OnlineSolver solver
   , goalSym ~ OnlineBackend s goalSolver fs
   , OnlineSolver goalSolver
-  , HasModel personality
   , ?outputConfig :: OutputConfig
   ) =>
   goalSym ->
   CruxOptions ->
   SimCtxt personality sym p ->
-  (Maybe (GroundEvalFn s) -> LPred sym ast -> IO (Doc Void)) ->
-  Maybe (Goals (LPred sym asmp) (LPred sym ast)) ->
-  IO (ProcessedGoals, Maybe (Goals (LPred sym asmp) (LPred sym ast, ProofResult (Either (LPred sym asmp) (LPred sym ast)))))
+  (Maybe (GroundEvalFn s) -> Assertion sym -> IO (Doc Void)) ->
+  Maybe (Goals (Assumptions sym) (Assertion sym)) ->
+  IO (ProcessedGoals, Maybe (Goals (Assumptions sym) (Assertion sym, [ProgramLoc], ProofResult sym)))
 
 proveGoalsOnline _ _opts _ctxt _explainFailure Nothing =
      return (ProcessedGoals 0 0 0 0, Nothing)
 
-proveGoalsOnline sym opts ctxt explainFailure (Just gs0) =
+proveGoalsOnline sym opts _ctxt explainFailure (Just gs0) =
```

Comment:
Can this `_ctxt` argument be removed?

Reply:
Not without changing the type of `ProverCallback`.  I think I'll leave it as-is for now.